src/base/QXmppIq.cpp

```diff
@@ -74,6 +74,18 @@ bool QXmppIq::isXmppStanza() const
     return true;
 }
 
+///
+/// Serializes the object's root XML element.
+///
+/// \param writer writer serializing the object
+///
+void QXmppIq::rootToXml(QXmlStreamWriter *writer) const
```

Comment:
Why do you need such a function. The use case you're describing is exactly what toXml() does.

Reply:
Only the root elements are serialized here. What exactly should be improved?

Reply:
fixed by not reserializing the IQ